editor/editor_themes.cpp

```diff
@@ -233,37 +248,50 @@ Ref<Theme> create_editor_theme(const Ref<Theme> p_theme) {
 	int LIGHT_COLOR = 2;
 	bool dark_theme = (icon_font_color_setting == AUTO_COLOR && ((base_color.r + base_color.g + base_color.b) / 3.0) < 0.5) || icon_font_color_setting == LIGHT_COLOR;
 
-	Color dark_color_1 = base_color.linear_interpolate(Color(0, 0, 0, 1), contrast);
-	Color dark_color_2 = base_color.linear_interpolate(Color(0, 0, 0, 1), contrast * 1.5);
-	Color dark_color_3 = base_color.linear_interpolate(Color(0, 0, 0, 1), contrast * 2);
+	const Color dark_color_1 = base_color.linear_interpolate(Color(0, 0, 0, 1), contrast);
+	const Color dark_color_2 = base_color.linear_interpolate(Color(0, 0, 0, 1), contrast * 1.5);
+	const Color dark_color_3 = base_color.linear_interpolate(Color(0, 0, 0, 1), contrast * 2);
 
-	Color contrast_color_1 = base_color.linear_interpolate((dark_theme ? Color(1, 1, 1, 1) : Color(0, 0, 0, 1)), MAX(contrast, default_contrast));
-	Color contrast_color_2 = base_color.linear_interpolate((dark_theme ? Color(1, 1, 1, 1) : Color(0, 0, 0, 1)), MAX(contrast * 1.5, default_contrast * 1.5));
+	const Color background_color = dark_color_2;
 
-	Color font_color = dark_theme ? Color(1, 1, 1) : Color(0, 0, 0);
-	Color font_color_disabled = dark_theme ? Color(0.6, 0.6, 0.6) : Color(0.45, 0.45, 0.45);
+	// white (dark theme) or black (light theme), will be used to generate the rest of the colors
+	const Color mono_color = dark_theme ? Color(1, 1, 1) : Color(0, 0, 0);
 
-	Color separator_color = dark_theme ? Color(1, 1, 1, 0.1) : Color(0, 0, 0, 0.1);
+	const Color contrast_color_1 = base_color.linear_interpolate(mono_color, MAX(contrast, default_contrast));
+	const Color contrast_color_2 = base_color.linear_interpolate(mono_color, MAX(contrast * 1.5, default_contrast * 1.5));
 
-	Color tab_color = highlight_tabs ? base_color.linear_interpolate(font_color, contrast) : base_color;
-	const int border_width = CLAMP(border_size, 0, 3) * EDSCALE;
+	const Color font_color = mono_color.linear_interpolate(base_color, 0.25);
```

Comment:
are you sure that we want the font color always to be 25% base color. So that all the fonts are coloured?

Reply:
no, i have to work on contrast for everything, but should be easier now